pkg/cmd/step/create/step_create_meta_pipeline.go

```diff
@@ -291,11 +267,46 @@ func (o *StepCreatePipelineOptions) getClientsAndNamespace() (tektonclient.Inter
 	return tektonClient, jxClient, kubeClient, ns, nil
 }
 
-func (o *StepCreatePipelineOptions) buildPullRevs(pullRefs string) (*prow.PullRefs, error) {
-	var pr *prow.PullRefs
-	var err error
-	if pullRefs != "" {
-		pr, err = prow.ParsePullRefs(pullRefs)
+// determineBranchIdentifier finds a identifier for the branch which is build by the specified pull ref.
+// The name is either an actual git branch name (eg master) or a synthetic names  like PR-<number> for single pull requests
+// or 'batch' for Prow batch builds.
+// The method makes the decision purely based on the Prow pull ref. At this stage we don't have the full ProwJov spec
+// available.
+func (o *StepCreatePipelineOptions) determineBranchIdentifier(pullRef prow.PullRefs) string {
```

Comment:
So all of this locks in to only kicking off builds via prow - that's not going to work. `jx start pipeline` is valid, e.g.

Reply:
`jx start pipeline` does create a Prow job if prow is enabled and goes into a different branch if not - https://github.com/jenkins-x/jx/blob/65962ff5ef1a6d1c4776daee0163434c9c2cb566/pkg/cmd/start/start_pipeline.go#L156

I think we are fine there.

Reply:
BTW, this happens in similar form in pipeline runner - https://github.com/jenkins-x/jx/blob/fe66ed51204667e37ed067f65a7ebc67f7c75bbb/pkg/cmd/controller/controller_pipelinerunner.go#L322

Reply:
Ok, then we should be fine there.